File: python/semanticabi/metadata/EthLog.py
from typing import List, TypedDict


class EthLog(TypedDict):
    address: str
    topics: List[str]
    data: str
    blockNumber: int
    transactionHash: str
    transactionIndex: int
    blockHash: str
    # log index relative to the entire block vs transaction possibly in hex
    logIndex: int | str
    removed: bool

File: js/src/semanticabi/metadata/EthLog.ts
export type EthLog = {

    address: string,
    topics: string[],
    data: string,
    blockNumber: number | string,
    transactionHash: string,
    transactionIndex: number | string,
    blockHash: string,
    //log index relative to the entire block vs transaction possibly in hex
    logIndex: number | string,
    removed: boolean

}